packages/ckeditor5-ui/src/dialog/dialogview.ts

```diff
@@ -0,0 +1,635 @@
+
+	/**
+	 * TODO
+	 */
+	private _getCurrentDomRoot: () => HTMLElement;
+
+	/**
+	 * TODO
+	 */
+	private _getViewportOffset: () => EditorUI[ 'viewportOffset' ];
+
+	/**
+	 * TODO
+	 */
+	declare public position: DialogViewPosition;
+
+	/**
+	 * TODO
+	 */
+	private readonly focusables: ViewCollection;
+
+	/**
+	 * TODO
+	 */
+	private readonly focusCycler: FocusCycler;
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( locale: Locale, getCurrentDomRoot: () => HTMLElement, getViewportOffset: () => EditorUI[ 'viewportOffset' ] ) {
+		super( locale );
+
+		const bind = this.bindTemplate;
+
+		this.set( 'isVisible', false );
+		this.set( 'className', '' );
+		this.set( 'isDraggable', false );
+		this.set( 'isTransparent', false );
+		this.set( 'wasMoved', false );
+		this.set( 'position', DialogViewPosition.SCREEN_CENTER );
+		this.set( '_top', 0 );
+		this.set( '_left', 0 );
+		this._getCurrentDomRoot = getCurrentDomRoot;
+		this._getViewportOffset = getViewportOffset;
+
+		this.decorate( 'moveTo' );
+
+		this.children = this.createCollection();
+		this.children.on<CollectionChangeEvent>( 'change', this._updateFocusCycleableItems.bind( this ) );
+
+		this.parts = this.createCollection();
+		this.contentView = this._createContentView();
+
+		this.keystrokes = new KeystrokeHandler();
+		this.focusTracker = new FocusTracker();
+		this.focusables = new ViewCollection();
+		this.focusCycler = new FocusCycler( {
+			focusables: this.focusables,
+			focusTracker: this.focusTracker,
+			keystrokeHandler: this.keystrokes,
+			actions: {
+				// Navigate form fields backwards using the Shift + Tab keystroke.
+				focusPrevious: 'shift + tab',
+
+				// Navigate form fields forwards using the Tab key.
+				focusNext: 'tab'
+			}
+		} );
+
+		this.setTemplate( {
+			tag: 'div',
+			attributes: {
+				class: [
+					'ck',
+					'ck-dialog-overlay',
+					bind.if( 'isDraggable', 'ck-dialog-overlay__transparent' ),
+					bind.if( 'isVisible', 'ck-hidden', value => !value )
+				],
+				// Prevent from editor losing focus when clicking on the modal overlay.
+				tabindex: '-1'
+			},
+			children: [
+				{
+					tag: 'div',
+					attributes: {
+						tabindex: '-1',
+						class: [
+							'ck',
+							'ck-dialog',
+							bind.if( 'isDraggable', 'ck-dialog_draggable' ),
+							bind.to( 'className' )
+						],
+						style: {
+							top: bind.to( '_top', top => toPx( top ) ),
+							left: bind.to( '_left', left => toPx( left ) ),
+							visibility: bind.if( 'isTransparent', 'hidden' )
+						}
+					},
+					children: this.parts
+				}
+			]
+		} );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override render(): void {
+		super.render();
+
+		this._updateFocusCycleableItems();
+
+		this.keystrokes.set( 'Esc', ( data, cancel ) => {
+			this.fire<DialogViewCloseEvent>( 'close' );
+			cancel();
+		} );
+
+		// Support for dragging the modal.
+		// TODO: Don't allow dragging beyond the edge of the viewport.
+		// TODO: Disable dragging when the mobile view is on.
```

Comment:
I am not sure how we can check "if the mobile view is on" right now + I think we decided that for now we don't address mobile -- as long as the UX does not degrade.